packages/go/openapi/src/schemas/model.auth-provider.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Specter Ops, Inc.
+#
+# Licensed under the Apache License, Version 2.0
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+type: object
+properties:
+  id:
+    type: integer
+    format: int32
+    description: Unique identifier for the SSO provider
+  name:
+    type: string
+    description: Name of the SSO provider
+  type:
+    type: string
+    description: Type of SSO provider (SAML or OIDC)
+  slug:
+    type: string
+    description: URL-friendly identifier for the provider
+  details:
```

Comment:
I think for this you can use `oneOf` which will let `details` be equal to one or the other of your defined schemas:
https://swagger.io/docs/specification/v3_0/data-models/oneof-anyof-allof-not/#oneof

And since the `details` are only used in this object, you should be fine defining it inline